bitcoin/src/amount.rs

```diff
@@ -83,8 +83,28 @@ impl Denomination {
             Denomination::MilliSatoshi => "msat",
         }
     }
+
+    /// The different str forms of denominations that are recognized.
+    fn forms(s: &str) -> Option<Self> {
+        match s {
+            "BTC" | "btc" | "BTCs" | "btcs" => Some(Denomination::Bitcoin),
+            "cBTC" | "cbtc" | "cBTCs" | "cbtcs" => Some(Denomination::CentiBitcoin),
+            "mBTC" | "mbtc" | "mBTCs" | "mbtcs" => Some(Denomination::MilliBitcoin),
+            "uBTC" | "ubtc" | "uBTCs" | "ubtcs" => Some(Denomination::MicroBitcoin),
+            "nBTC" | "nbtc" | "nBTCs" | "nbtcs" => Some(Denomination::NanoBitcoin),
+            "pBTC" | "pbtc" | "pBTCs" | "pbtcs" => Some(Denomination::PicoBitcoin),
+            "bit" | "bits" => Some(Denomination::Bit),
+            "SATOSHI" | "Satoshi" | "Satoshis" | "satoshi" | "satoshis" | "Sat" | "sat" | "SAT"
+            | "sats" => Some(Denomination::Satoshi),
+            "mSAT" | "msat" | "mSATs" | "msats" => Some(Denomination::MilliSatoshi),
+            _ => None,
+        }
+    }
 }
 
+const CONFUSING_FORMS: [&str; 9] =
+    ["Msat", "Msats", "MSAT", "MSATS", "MSat", "MSats", "MBTC", "Mbtc", "PBTC"];
```

Comment:
```suggestion
const CONFUSING_FORMS: [&str; 4] = ["MSAT", "MSATS", "MBTC", "PBTC"];
```
I would argue that only "M" for mega and "P" for peta are confusing, "Msat" is not confusing because since the "M" is capitalized while the rest is not, the author meant "mega" which we do not support.

Reply:
FTR I say `Msat` commonly - usually in relation to channel sizes. But I'm still kinda uncomfortable parsing it here.

Reply:
I don't really see the value in keeping CONFUSING_FORMS at all.  Although I can update it to the four you suggest.

Reply:
"MSAT is not allowed because M could mean mega but many people use it as milli" is infinitely better error message than "unrecognized denomination". Just compare Rust error messages and PHP and you should be able to see why it matters. :)